index.js

```diff
@@ -0,0 +1,224 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+const yaml = require('js-yaml')
+const path = require('path')
+
+function fastifySwagger (fastify, opts, next) {
+  fastify.decorate('swagger', swagger)
+
+  opts = opts || {}
+  opts.swagger = opts.swagger || {}
+
+  const info = opts.swagger.info || null
+  const host = opts.swagger.host || null
+  const schemes = opts.swagger.schemes || null
+  const consumes = opts.swagger.consumes || null
+  const produces = opts.swagger.produces || null
+  const basePath = opts.swagger.basePath || null
+
+  const exposeRoute = !(opts.exposeRoute === false)
+  if (exposeRoute) {
+    fastify.get(opts.route || '/documentation', {
+      schema: {
+        description: 'Documentation route, if yaml is true the payload will be in yaml format',
+        querystring: {
+          type: 'object',
+          properties: {
+            yaml: { type: 'boolean' }
+          }
+        },
+        response: {
+          200: {
+            description: 'The swagger definition of the routes',
+            type: 'object',
+            properties: {
+              hello: { type: 'string' }
+            }
+          }
+        }
+      }
+    }, function (req, reply) {
+      if (req.query.yaml) {
+        return reply
+          .header('Content-Type', 'text/plain')
```

Comment:
I would go for `application/x-yaml` with an explanation and a link to the IANA registry.